hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -1972,6 +1972,9 @@ public TableGroupJoin createTableGroupJoin(
 							true,
 							creationState
 					);
+					if ( getAssociatedEntityMappingType().getSuperMappingType() != null && !creationState.supportsEntityNameUsage() ) {
+						getAssociatedEntityMappingType().applyDiscriminator( p -> {}, null, tableGroup, creationState );
```

Comment:
As I mentioned you can just pass `null` instead of an empty lambda as predicate consumer:
```suggestion
						getAssociatedEntityMappingType().applyDiscriminator( null, null, tableGroup, creationState );
```